webapp/channels/src/components/announcement_bar/text_dismissable_bar.tsx

```diff
@@ -20,67 +20,57 @@ interface Props extends Partial<AnnouncementBarProps> {
     className?: string;
 }
 
-type State = {
-    dismissed: boolean;
-}
-
-export default class TextDismissableBar extends React.PureComponent<Props, State> {
-    constructor(props: Props) {
-        super(props);
-
-        this.state = {
-            dismissed: true,
-        };
-    }
+const TextDismissableBar = (props: Props) => {
+    const {allowDismissal, text, ...extraProps} = props;
+    const [dismissed, setDismissed] = useState<boolean>(false);
 
-    static getDerivedStateFromProps(props: Props) {
-        const dismissed = localStorage.getItem(localStoragePrefix + props.text?.toString());
-        return {
-            dismissed: (dismissed === 'true'),
-        };
-    }
+    useEffect(() => {
+        const dismissedFromStorage = localStorage.getItem(localStoragePrefix + text?.toString());
+        if (dismissedFromStorage === 'true') {
+            setDismissed(true);
+        }
+    }, [text]);
 
-    handleDismiss = () => {
-        if (!this.props.allowDismissal) {
+    const handleDismiss = () => {
```

Comment:
handleDismiss is used as a prop, so it should use a `useCallback` to not pass a new function on every render.